### ISSUE TITLE
Exact Search on Rule Name

### ISSUE THREAD

**Issue by jcbowman** [Yelp/elastalert#1789]
In seems like in Elasticsearch 5.0+ we should use the '.keyword' modifier when pulling the elastalert metadata out of elasticsearch. Without the keyword modifier the metadata query will never match rule_name that have spaces, uppercase letters, etc because the analyzer will strip those out.

For example this line:

https://github.com/Yelp/elastalert/blob/0465d6b739870c0fce148de0554018a10c9f0019/elastalert/elastalert.py#L1742

Should be something like this is elasticsearch 5.0+:
`query = {'term': {'rule_name.keyword': rule_name}}`

I came across this because I was having issues with the realert feature and realized on restart elastalert was failing to find the metadata for silenced rules because our rule_name had spaces in it.

**Comment by jcbowman** [comment]
Good point - I missed that. It looks like it should already be setup as a keyword mapping.

Our mapping is setup to use both keyword and text though - I'm not sure how we would have gotten into this situation. Any ideas?

```json
"silence": {
	"properties": {
		"@timestamp": {
			"type": "date"
		},
		"exponent": {
			"type": "long"
		},
		"rule_name": {
			"type": "text",
			"fields": {
				"keyword": {
					"type": "keyword",
					"ignore_above": 256
				}
			}
		},
		"until": {
			"type": "date"
		}
	}
} 
```

**Comment by jcbowman** [comment]
Actually. I think I might know what happened. 

If something was running elastalert and the index it was writing it's metadata too got deleted for any reason the data coming from elastalert would recreate the index and use default types for all of the attributes. Does that seem right?

**Comment by Qmando** [comment]
Yeah, that does sounds correct for what would happen if the index was deleted while elastalert was running. I'd expect you might see errors or maybe not, depending on what part of the code was running at that moment.

**Comment by jcbowman** [comment]
I'm closing this issue. I deleted our index and let elastalert recreate it and it seems to have fixed our issue. Thanks for your help!

**Comment by jcbowman** [comment]
@Qmando It looks like the same issue is still occurring. I think I might have spoke too soon saying it fixed the problem.

Looking at this closer shouldn't it be `elasticversion >= 5`
https://github.com/Yelp/elastalert/blob/master/elastalert/create_index.py#L133
